src/store/modules/user/actions.ts

```diff
@@ -130,13 +130,15 @@ const actions: ActionTree<UserState, RootState> = {
   /**
    * update current eComStore information
    */
-  async setEcomStore({ commit, dispatch }, payload) {
+  async setEcomStore({ commit, dispatch, state }, productStore) {
+    const currentEComStore = state.currentEComStore;
+    const productStoreId = typeof productStore === 'string' ? productStore : productStore.productStoreId;
```

Comment:
I would suggest we check if payload has productStoreId we find the productStore and if productStore we set it as is

Reply:
Yes, sir, will change it as stated.